### PR TITLE
feat: trending videos enhancements

### PR DESCRIPTION
**Pull request by ice-endymion**
## Description
The goal of this PR is to mimic `PostEntity` data for testing and development purposes. Key updates include:
- New fake post and video generation logic.
- Improved `TrendingVideoListItem` with safer null handling and better readability.
- Updates to video page components and routing for smoother navigation.
- Enhanced placeholder handling for missing video thumbnails.

## Additional Notes
- Addressed single-line text behavior in `VideoTextPost`.
- Added mock data to simulate `PostEntity` while real data implementation is pending.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactoring
- [ ] Documentation
- [ ] Chore

## Screenshots
https://github.com/user-attachments/assets/f29e27ac-314d-469a-bf29-40662b221454

## Figma Link
Do not pay attention to the inconsistency of the design, video player screen will be modified in the following PR.
[Design](https://www.figma.com/design/3TnZngoFeklOVC8TexlR1D/%F0%9F%A7%8A-ICE-Cloud?node-id=18296-116569&t=aLU2hgOl39kAl4iT-4)

### CODE REVIEW

**Review by ice-kreios — DISMISSED** [review]

File: lib/app/features/feed/views/pages/feed_page/components/trending_videos/trending_videos.dart [hunk 1]
```diff
@@ -22,7 +23,11 @@ class TrendingVideos extends ConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     final listOverlay = ref.watch(trendingVideosOverlayNotifierProvider);
     final dataSource = ref.watch(feedTrendingVideosDataSourceProvider);
-    final videosData = ref.watch(entitiesPagedDataProvider(dataSource));
+    // TODO: remove this [mockPostEntitiesPagedDataProvider] when we have real data
+    // final videosData = ref.watch(entitiesPagedDataProvider(dataSource));
+    final videosData = ref.watch(mockPostEntitiesPagedDataProvider(dataSource));
+
+    if (videosData?.data.items.isEmpty ?? true) return const SizedBox.shrink();
```
[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/features/feed/views/pages/feed_page/components/trending_videos/trending_videos.dart:30]: Hmm, test it, looks bad if we got empty list, for empty list we can hide videos with `return const SizedBox.shrink();`, for null we can show `Skeleton`, but can we have case when we got only null?
Screenshot 2024-11-27 at 12 47 33

File: lib/app/features/video/views/components/video_post_text.dart [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  Widget build(BuildContext context, WidgetRef ref) {
+    final postMedia = usePostMedia(postEntity.data);
+    final isTextExpanded = useState(false);
+
+    final postText = extractPostText(
+      postEntity.data.content,
+      excludeMatcherType: UrlMatcher,
+    );
+
+    final isOneLine = _isTextOneLine(
```
Comment (ice-tychon) [inline lib/app/features/video/views/components/video_post_text.dart:45]: Please, cache the result

File: lib/app/features/video/views/components/video_post_text.dart [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      text: postText,
+      style: context.theme.appTextThemes.body2.copyWith(color: Colors.white),
+      maxWidth: MediaQuery.of(context).size.width,
```
Comment (ice-alcides) [inline lib/app/features/video/views/components/video_post_text.dart:48]: use `MediaQuery.sizeOf(context).width`

File: lib/app/features/feed/views/components/post/components/post_body/post_body.dart [hunk 4]
```diff
@@ -20,19 +21,14 @@ class PostBody extends ConsumerWidget {
...
     final postMedia = usePostMedia(postEntity.data);
 
-    //TODO::temp impl
-    final postText = postEntity.data.content.fold<StringBuffer>(StringBuffer(), (result, match) {
-      if (match.matcherType != UrlMatcher) {
-        result.write(match.text);
-      }
-      return result;
-    });
+    final postText = extractPostText(postEntity.data.content, excludeMatcherType: UrlMatcher);
```
[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/features/feed/views/components/post/components/post_body/post_body.dart:24]: I will do it in separate PR which introduces new common Service for getting TextSpan from TextMatch.

File: lib/app/features/feed/views/pages/feed_page/components/trending_videos/components/trending_video_likes_button.dart [hunk 5]
```diff
@@ -34,15 +34,16 @@ class TrendingVideoLikesButton extends ConsumerWidget {
             size: 14.0.s,
             color: context.theme.appColors.secondaryBackground,
           ),
-          Padding(
-            padding: EdgeInsets.only(left: 2.0.s),
-            child: Text(
-              formatDoubleCompact(likesCount ?? 0),
-              style: context.theme.appTextThemes.caption3.copyWith(
-                color: context.theme.appColors.secondaryBackground,
+          if (likesCount != null)
```
Comment (ice-kreios) [inline lib/app/features/feed/views/pages/feed_page/components/trending_videos/components/trending_video_likes_button.dart:37]: should we check also for eq to zero?

File: lib/app/features/feed/views/components/user_info/user_info.dart [hunk 6]
```diff
@@ -36,8 +37,18 @@ class UserInfo extends HookConsumerWidget {
         }
         return ListItem.user(
           onTap: () => ProfileRoute(pubkey: pubkey).push<void>(context),
-          title: Text(userMetadataEntity.data.displayName),
-          subtitle: Text(prefixUsername(username: userMetadataEntity.data.name, context: context)),
+          title: Text(
+            userMetadataEntity.data.displayName,
+            style: TextStyle(
```
Comment (ice-kreios) [inline lib/app/features/feed/views/components/user_info/user_info.dart:42]: i think we should  apply style with `context.theme.appTextThemes`

File: lib/app/features/feed/views/components/user_info/user_info.dart [hunk 7]
```diff
@@ -36,8 +37,18 @@ class UserInfo extends HookConsumerWidget {
...
+              color: textColor,
+            ),
+          ),
+          subtitle: Text(
+            prefixUsername(username: userMetadataEntity.data.name, context: context),
+            style: TextStyle(
```
Comment (ice-kreios) [inline lib/app/features/feed/views/components/user_info/user_info.dart:48]: i think we should  apply style with `context.theme.appTextThemes`

File: lib/app/features/feed/views/pages/feed_page/components/trending_videos/components/trending_videos_list_item.dart [hunk 8]
```diff
@@ -18,52 +16,110 @@ class TrendingVideoListItem extends ConsumerWidget {
...
+
+  final Size size;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: size.width,
+      height: size.height,
+      decoration: BoxDecoration(
+        color: Colors.grey.withOpacity(0.3),
```
Comment (ice-kreios) [inline lib/app/features/feed/views/pages/feed_page/components/trending_videos/components/trending_videos_list_item.dart:58]: i think we should use `context.theme.appColors`

File: lib/app/features/feed/views/components/user_info/user_info.dart (hunk 6 above)
Comment (ice-endymion) [reply lib/app/features/feed/views/components/user_info/user_info.dart:42]: I pass optional `textColor` param to TextStyle, and it is used only to show white colors text on VideoPage black background. I think it is better to pass TextStyle? itself.

File: lib/app/features/feed/views/components/user_info/user_info.dart (hunk 7 above)
Comment (ice-endymion) [reply lib/app/features/feed/views/components/user_info/user_info.dart:48]: Same here.

**Review by ice-kreios — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

**Review by ice-tychon — APPROVED** [review]

**Pull request merged**